public/src/js/models/collections/collection.js

```diff
@@ -234,29 +236,30 @@ export default class Collection extends BaseClass {
 
         authedAjax.request({
             type: 'post',
-            url: vars.CONST.apiBase + '/collection/' + (goLive ? 'publish' : 'discard') + '/' + this.id,
+            url: `${vars.CONST.apiBase}/collection/${action}/${this.id}`,
             data: requestData ? JSON.stringify(requestData) : undefined
         })
+        .then(() => this.load().then(detectPressFailures))
+        .then(() => {
+            if (opts.sendAlert) {
+                success(requestData);
+            }
+        })
         .catch(error => {
             const errorMessages = [];
             try {
                 errorMessages.push.apply(errorMessages, JSON.parse(error.responseText));
             } catch (ex) {
                 errorMessages.push(error.responseText || error.message);
             }
-            const message = 'Error when ' + (goLive ? 'publishing' : 'discarding')
-                + ' the collection: ' + errorMessages.join('<br>');
-            alert(message);
+
+            const message = `Error when ${action}ing the collection: ${errorMessages.join('<br>')}`;
             reportErrors(new Error(message));
         })
-        .catch(() => {})
-        .then(() => this.load().then(detectPressFailures))
-        .then(() => {
-            if (opts.sendAlert) {
-                success(requestData);
-            }
-        })
-        .catch(() => {});
+        .catch(err => {
+            this.setPending(false);
+            alert(err.message);
```

Comment:
I think it would be good to have a custom client facing text here without any of the errors which are coming back from the server. Breaking news collections should have a different message from normal collections as well from normal collections just to make it clear what is happening. Calling `confirmSendingAlert()` will tell you if we are dealing with a breaking news collection.

Reply:
Updated now. In two minds about whether the server response should go to sentry or not, I think its useful if it does.

Reply:
I think it might as well although we log it on server side as well